src/score_hv/harvesters/gsi_satellite_radiance.py

```diff
@@ -0,0 +1,415 @@
+                
+                data_usage_dict = dict()
+                for var in self.config.vars_to_harvest:
+                    if var == 'var':
+                        value = float(line2list[2].split()[0])
+                    elif var == 'varch_cld':
+                        value = float(line2list[3].split()[0])
+                    elif var == 'use':
+                        value = int(line2list[4].split()[0])
+                    elif var == 'ermax':
+                        value = float(line2list[5].split()[0])
+                    elif var == 'b_rad':
+                        value = float(line2list[6].split()[0])
+                    elif var == 'pg_rad':
+                        value = float(line2list[7].split()[0])
+                    elif var == 'icld_det':
+                        value = int(line2list[8].split()[0])
+                    elif var == 'icloud':
+                        value = int(line2list[9].split()[0])
+                    elif var == 'iaeros':
+                        value = int(line2list[10].split()[0])
+                    elif var == BIAS_CORR_COEF_STR:
+                        value = list() # empty list for bias
+                                       # correction coeficients
+                    
+                    data_usage_dict[var] = value    
+                
+                self.channels[series_number] = {
+                    'observation_type': observation_type,
+                    'channel': channel,
+                    'data_usage_dict': data_usage_dict
+                }
+                
+                # store channel numbers
+                if observation_type in self.obs_type_channels.keys():
+                    self.obs_type_channels[observation_type].append(
+                        channel)
+                else:
+                    self.obs_type_channels[observation_type] = [channel]
+                    
+                # store series numbers
+                if observation_type in self.obs_type_series_numbers.keys():
+                    self.obs_type_series_numbers[observation_type].append(
+                        series_number)
+                else:
+                    self.obs_type_series_numbers[observation_type] = [
+                        series_number]
+                
+                if series_number == self.nchannels:
+                    """this is the last channel
+                    """
+                    radinfo_read1 = False
+                
+            elif radinfo_read2:
+                """RADINFO_READ PART 2
+                
+                harvest bias correction coeficients
+                """
+
+                series_number = int(line_parts[0])
+                #channel_index = series_number - 1
+                
+                # partial assurance that this is the correct channel, by name
+                assert line_parts[1] == self.channels[series_number]['observation_type']
```

Comment:
I'm wondering if this is a proper place to be using 'assert' since the comment above says the purpose is 'partial assurance'. Using assert is a very strong forceful piece of code, so the use case should be for "strong confirmed assurance that this is the correct channel by name"; i.e. that the code cannot possibly continue if this line is False. I don't feel I know this code well enough to say whether that's the case here and that the code should crash at this point if it's not True. is that the case?

Reply:
The code should definitely crash if this is not true. This assert statement is a quick run time check representing "partial assurance" because it alone is not enough to test all the corner cases. More thorough unit tests are provided in the test file.

Reply:
replaced with if block -> raise ValueErrors if false